lib/pbench/server/api/resources/metadata_api.py

```diff
@@ -0,0 +1,355 @@
+            abort(400, message="Invalid json object in request")
+
+        try:
+            metadata_session = Metadata.query(id=id)[0]
+        except Exception:
+            self.logger.exception(
+                "Exception occurred in the PUT request while querying the Metadata model, id: {}",
+                id,
+            )
+            abort(500, message="INTERNAL ERROR")
+
+        # Verify if the metadata session id in the url belongs to the logged in user
+        self.verify_metadata(metadata_session, "put")
+
+        # Check if the metadata payload contain fields that are either protected or
+        # not present in the metadata db. If any key in the payload does not match
+        # with the column name we will abort the update request.
+        non_existent = set(post_data.keys()).difference(
+            set(Metadata.__table__.columns.keys())
+        )
+        if non_existent:
+            self.logger.warning(
+                "User trying to update fields that are not present in the metadata database. Fields: {}",
+                non_existent,
+            )
+            abort(400, message="Invalid fields in update request payload")
+        protected = set(post_data.keys()).intersection(set(Metadata.get_protected()))
+        for field in protected:
+            if getattr(metadata_session, field) != post_data[field]:
+                self.logger.warning(
+                    "User trying to update the non-updatable fields. {}: {}",
+                    field,
+                    post_data[field],
+                )
+                abort(403, message="Invalid update request payload")
+        try:
+            metadata_session.update(**post_data)
+            self.logger.info(
+                "User metadata session updated, id: {}", metadata_session.id
+            )
+        except Exception:
+            self.logger.exception("Exception occurred updating the Metadata model")
+            abort(500, message="INTERNAL ERROR")
+        else:
+            response_object = {
+                "message": "success",
+                "data": metadata_session.get_json(),
+            }
+            return make_response(jsonify(response_object), 200)
+
+    @Auth.token_auth.login_required()
+    def delete(self, id=None):
+        """
+        Delete request for deleting a metadata session of a user given a metadata id.
+        This requires a Pbench auth token in the header field
+
+        The url requires a metadata session id such as /user/metadata/<int:id>
+
+        Required headers include
+            Authorization:   Bearer <Pbench authentication token (user received upon login)>
+
+        :return: JSON Payload
+            response_object = {
+                    "message": "success"
+                }
+        """
+        if not id:
+            self.logger.warning("Metadata id not provided during metadata query")
+            abort(400, message="Please provide a metadata id to query")
```

Comment:
Again, I like "metadata id" here instead of "session" ... but this isn't the query API, we're trying to delete it.